go/vt/sqlparser/testdata/union_cases.txt

```diff
@@ -1004,7 +1004,7 @@ INPUT
 SELECT 1 FOR SHARE UNION SELECT 2;
 END
 ERROR
-syntax error at position 19 near 'SHARE'
```

Comment:
this actually matches the error mysql8 provides when issuing the same query via explain:

```
mysql> explain select 1 for share union select 2;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'union select 2' at line 1
```